trigger.js

```diff
@@ -206,6 +215,44 @@ class TriggerHappy {
         Hooks.once('updateToken', () => this._executeTriggers(triggers));
         return true;
     }
+  
+    _onPreUpdateToken(scene, id, update) {
+        if (!scene.isView) return true;
+        if (update.x === undefined && update.y === undefined) return true;
+        const token = scene.data.tokens.find(t => t._id === update._id);
+        if (token.hidden) return true; // don't stop ivnisible tokens?
+
+        // Get all trigger tokens in scene
+        let targets = this._getTriggerTokens(canvas.tokens.placeables, this.triggers, 'capture');
+        if (!targets) return true;
+
+        let x = update.x || token.x;
+        let  y = update.y || token.y;
+        let tw = token.width * canvas.scene.data.grid / 2;
+        let th = token.height * canvas.scene.data.grid / 2;
+        let motion = new Ray({x: token.x + tw, y: token.y + th}, {x: x + tw, y: y + th});
+
+        // don't trigger on tokens that are already captured
+        targets = targets.filter(target => token.x !== target.x + target.w  / 2 - tw || token.y !== target.y + target.h / 2 - th)
+        
+        // sort list by distance from start token position
+        targets.sort((a , b) => dist(token.x, token.y, b.x, b.y) - dist(token.x, token.y, a.x, a.y));
+        
+        targets.forEach(target => {
+            // test motion vs token diagonals
+            if (motion.intersectSegment([target.x, target.y, target.x + target.w, target.y + target.h])
+            || motion.intersectSegment([target.x, target.y + target.h, target.x + target.w, target.y])) {

```

Comment:
Nice work on the intersect. So if the movement line intersects the X around the target, it will trigger the capture and move the token to the center of the target.
This works mostly, but I see two problematic situations :
- Imagine a 9x9 trap, if the token move into one of the 4 grid squares at the N-S-E-W coordinates (not the corners or the center grid), then they won't intersect the X line you built here. Solution for that would be to intersect with the 4 lines `[x, y, x + w, y], [x + y, y, x + w, y +h], [x + w, x + h, x, y +h], [x, y + h, x, y]`
- Same 9x9 trap, using either current code or the 4 intersections I suggested above, token crosses one of the lines diagonally, they get transported to the center of the trap, even though it wasn't on their path. I have no solution for that, and I guess it's an acceptable situation since it's a "capture" and so it captures the token to its center. As long as it's documented, it should be fine.

Reply:
Problem 1, Good call. I was hoping to avoid doing 4 intersects per token, since it is already quite expensive and done for all trigger tokens on the map on every token call. It's probably worth it, computationally, to check the size of the trigger token and do 4 lines for tokens bigger than 2 * 2
Problem 2. Agree, but I don't have a good solution The only thing I came up with is checking each square of the token, sorted by distance, and  then placing the token there. But I decided it was not worth the cpu cycles.
Another possibility. Once captured compute the closes square covered by the target to the token start and placing the token on that square. It will make the filter to move out of a token more expensive but it could be treated like a move onto token test. I'll have a look at this

Reply:
I think I will leave it the way it is for now (who has such big pits anyway) and investigate doing the multi square target thing at the same time.

Reply:
My bad, I pointed out the potential problems but didn't say how strongly I feel about them. I think that they are mostly acceptable situations. I hadn't thought of doing the 4 intersects on 3x3+ tokens only. That would be a pretty good improvement indeed. And the second issue is not really a problem, if someone doesn't like that a "capture" trap captures the token to its center, then can always just make multiple smaller traps if needed. Leave that up to the user rather than to the module to do the work.
I don't think checking every square is worth it, so I'd say don't bother. Maybe just do the 3x3+ tokens using 4 intersects and keep it as 2 intersects for smaller than 3x3 (not sure if 3x2 counts?).

Reply:
In fact problem 1 is not a problem at all. Since the capture trigger just stops the motion, the @Token trigger will fire anyway since the token ends up inside the target token. I knew I had thought about this later.

Reply:
ouhhh, nice, that's because capture triggers aren't mutually exclusive, so a capture trigger is also going to be a move trigger (usually.. if not, then for big traps, it would be the user's responsability to set both types), nice thinking! :D